benchmark.py

```diff
@@ -0,0 +1,203 @@
+    )
+    bc_outgas = dirichletbc(Constant(my_mesh, PETSc.ScalarType(0)), right_dofs, V)
+    bcs = [bc_sieverts, bc_outgas]
+
+    D_0 = 1.9e-7
+    E_D = 0.2
+    D = D_0 * exp(-E_D / k_B / temperature)
+
+    dt = 1 / 20
+    final_time = 50
+    num_steps = int(final_time / dt)
+
+    F = dot(D * grad(u), grad(v)) * dx(1)
+    F += ((u - u_n) / dt) * v * dx(1)
+
+    problem = NonlinearProblem(F, u, bcs=bcs)
+    solver = NewtonSolver(MPI.COMM_WORLD, problem)
+    solver.convergence_criterion = "incremental"
+    solver.rtol = 1e-10
+    solver.atol = 1e10
+    solver.report = True
+    ksp = solver.krylov_solver
+    opts = PETSc.Options()
+    option_prefix = ksp.getOptionsPrefix()
+    opts[f"{option_prefix}ksp_type"] = "cg"
+    opts[f"{option_prefix}pc_type"] = "gamg"
+    opts[f"{option_prefix}pc_factor_mat_solver_type"] = "mumps"
+    ksp.setFromOptions()
+
+    mobile_xdmf = XDMFFile(MPI.COMM_WORLD, "mobile_concentration.xdmf", "w")
+    mobile_xdmf.write_mesh(my_mesh)
+
+    flux_values = []
+    times = []
+    t = 0
+    progress = tqdm.autonotebook.tqdm(
+        desc="Solving H transport problem", total=num_steps
+    )
+    for i in range(num_steps):
+        progress.update(1)
+        t += dt
+
+        solver.solve(u)
+
+        surface_flux = form(D * dot(grad(u), n) * ds(2))
+        flux = assemble_scalar(surface_flux)
+        flux_values.append(flux)
+        times.append(t)
+
+        mobile_xdmf.write_function(u, t)
+
+        u_n.x.array[:] = u.x.array[:]
+
+    mobile_xdmf.close()
+
+    # analytical solution
+    S = S_0 * exp(-E_S / k_B / float(temperature))
+    permeability = float(D) * S
+    times = np.array(times)
+
+    n_array = np.arange(1, 10000)[:, np.newaxis]
+    summation = np.sum(
+        (-1) ** n_array * np.exp(-((np.pi * n_array) ** 2) * float(D) / L**2 * times),
+        axis=0,
+    )
+    analytical_flux = P_up**0.5 * permeability / L * (2 * summation + 1)
+
+    analytical_flux = np.abs(analytical_flux)
+    flux_values = np.array(np.abs(flux_values))
+
+    relative_error = np.abs((flux_values - analytical_flux) / analytical_flux)
+
+    relative_error = relative_error[
+        np.where(analytical_flux > 0.01 * np.max(analytical_flux))
+    ]
+    error = relative_error.mean()
+
+    assert error < 0.01
+
+
+def test_festim_vs_fenics_permeation_benchmark():
+    repetitions = 10
+
+    fenics_times = []
+    for i in range(repetitions):
+        start = time.time()
+        fenics_test_permeation_problem()
+        fenics_times.append(time.time() - start)
+    fenics_time = np.mean(fenics_times)
+
+    festim_times = []
+    for i in range(repetitions):
+        start = time.time()
+        test_permeation_problem()
+        festim_times.append(time.time() - start)
+    festim_time = np.mean(festim_times)
+
+    diff = (np.abs(fenics_time - festim_time) / ((fenics_time + festim_time) / 2)) * 100
+    if diff > 20:
+        raise ValueError(f"festim is {diff:.1f}% slower than fenics")
```

Comment:
It'd be nice to print it nonetheless
```suggestion
        raise ValueError(f"festim is {diff:.1f}% slower than fenics")
    else:
        print(f"festim is {diff:.1f}% slower than fenics")
```